R/checkers.R

```diff
@@ -0,0 +1,207 @@
+    "age groups should be contiguous" =
+      all(min(age_groups):max(age_groups) %in% age_groups),
+    "age groups should include only positive integers" =
+      checkmate::test_integerish(unlist(age_bounds), lower = 0)
+  )
+
+  # add and sort data frames cols
+  x <- data.frame(
+    min_age = vapply(age_bounds, "[[", FUN.VALUE = numeric(1), 1),
+    max_age = vapply(age_bounds, "[[", FUN.VALUE = numeric(1), 2),
+    proportion = x$proportion
+  )
+
+  # return age data frame
+  x
+}
+
+#' Check if arguments input to simulation function are valid
+#'
+#' @details Arguments that are used by all simulation functions are required
+#' and not given a default value, for other arguments that are not inputs in
+#' all simulation functions a default of `NULL` is used.
+#'
+#' Defaults mentioned in argument documentation is the default for the exported
+#' simulation function and not the default in this checking function. In this
+#' function there is either no default or `NULL`.
+#'
+#' @param sim_type A `character` string specifying which simulation function
+#' this function is being called within.
+#' @inheritParams sim_linelist
+#' @inheritParams sim_contacts
+#'
+#' @return Invisibly return the `sim_type` `character` string. The function is
+#' called for its side-effects, which will error if the input is invalid.
+#' @keywords internal
+.check_sim_input <- function(sim_type = c("linelist", "contacts", "outbreak"),
+                             contact_distribution,
+                             contact_interval,
+                             prob_infect,
+                             outbreak_start_date,
+                             min_outbreak_size,
+                             onset_to_hosp = NULL,
+                             onset_to_death = NULL,
+                             add_names = NULL,
+                             add_ct = NULL,
+                             case_type_probs = NULL,
+                             contact_tracing_status_probs = NULL,
+                             hosp_risk = NULL,
+                             hosp_death_risk = NULL,
+                             non_hosp_death_risk = NULL,
+                             population_age = NULL) {
+  sim_type <- match.arg(sim_type)
+
+  checkmate::assert_number(prob_infect, lower = 0, upper = 1)
+  .check_func_req_args(contact_distribution)
+  .check_func_req_args(contact_interval)
+  checkmate::assert_date(outbreak_start_date)
+  checkmate::assert_integerish(min_outbreak_size, lower = 1)
+
+  stopifnot(
+    "population_age must be two numerics or a data.frame" =
+      checkmate::test_numeric(population_age, len = 2) ||
+        is.data.frame(population_age)
+  )
+
+  if (sim_type %in% c("linelist", "outbreak")) {
+    .check_func_req_args(onset_to_hosp)
+    .check_func_req_args(onset_to_death)
+    checkmate::assert_logical(add_names, len = 1)
+    checkmate::assert_logical(add_ct, len = 1)
+    checkmate::assert_numeric(case_type_probs, len = 3)
```

Comment:
Suggest adding bounds checks for `case_type_probs`, as individual values could be negative and still sum to 1.0

Reply:
Resolved in #83.